src/internal/metrics/metrics.go

```diff
@@ -146,27 +146,31 @@ func FinishReportAndFlushUserAction(action string, err error, start time.Time) f
 	return wait
 }
 
-func (r *Reporter) reportClusterMetrics(ctx context.Context) {
+func (r *Reporter) reportClusterMetrics(rctx context.Context) {
 	ticker := time.NewTicker(reportingInterval)
 	defer ticker.Stop()
 	for {
 		select {
-		case <-ctx.Done():
+		case <-rctx.Done():
 			return
 		case <-ticker.C:
 		}
 		metrics := &Metrics{}
-		r.internalMetrics(metrics)
-		externalMetrics(r.env.GetKubeClient(), metrics) //nolint:errcheck
+		ctx, c := context.WithTimeout(rctx, reportingInterval/2)
```

Comment:
If all this is put into the ticker `case` of the `select` (which I think makes more sense anyway), the the scoping here is more obvious.